src/inference/src/dev/core_impl.cpp

```diff
@@ -57,8 +57,251 @@ void stripDeviceName(std::string& device, const std::string& substr) {
+                }
+                // 2.1.2. Device name is present in config file, merge properties according to:
+                // ov::device::properties(<device_name>) overrides ov::device::properties(ov::AnyMap{})
+                auto& secondary_device_properties = secondary_properties_it->second.as<ov::AnyMap>();
+                for (auto& item : secondary_property->second.as<ov::AnyMap>()) {
+                    secondary_device_properties[item.first] = item.second;
+                }
+            }
+        }
+
+        // 3. since the sub-property is flattened, we need to drop it
+        secondary_property = result_properties.erase(secondary_property);
+    }
+
+    // Second search for ov::device::properties(ov::AnyMap{...})
+    for (auto property = result_properties.begin(); property != result_properties.end();) {
+        if (property->first != ov::device::properties.name()) {
+            // 1. Skip non-matching properties
+            property++;
+            continue;
+        }
+
+        // 2. device properties DEVICE_PROPERTIES are found
+        if (property->second.is<std::string>()) {  // because of legacy API 1.0
+            property->second = property->second.as<ov::AnyMap>();
+        }
+        auto& secondary_properties = property->second.as<ov::AnyMap>();
+
+        for (auto secondary_property = secondary_properties.begin();
+             secondary_property != secondary_properties.end();) {
+            // flattening is performed only when config is applicable (see docs for ov::is_config_applicable)
+            if (ov::is_config_applicable(user_device_name, secondary_property->first)) {
+                // 2.1. flatten the secondary property for target device
+                // example: core.compile_model("GPU", ov::device::properties("GPU", ov::prop1));
+                // example: core.compile_model("GPU.1", ov::device::properties("GPU", ov::prop1));
+                update_result_properties(secondary_property->second.as<ov::AnyMap>());
+                secondary_property = secondary_properties.erase(secondary_property);
+            } else if (is_virtual_device(user_device_name)) {
+                // 2.2. keep the secondary property for the other virtual devices
+                secondary_property++;
+                continue;
+            } else {
+                // 2.3. remove the secondary property setting for other hardware device
+                // example: core.compile_model("GPU", ov::device::properties("CPU", ov::prop1));
+                secondary_property = secondary_properties.erase(secondary_property);
+            }
+        }
+
+        // 3. go to the next property
+        if (secondary_properties.empty()) {
+            // 3.1. since the sub-property is flattened, we need to drop it
+            property = result_properties.erase(property);
+        } else {
+            // 3.2. some properties are still in ov::device::properties(ov::AnyMap{}), abort loop
+            break;
```

Comment:
This branch means there are left properties for other virtual plugin? Why break looping other secondary_properties rather than better handling?

Reply:
> This branch means there are left properties for other virtual plugin? Why break looping other secondary_properties rather than better handling?

In this point secondary properties can be present one time only with key `DEVICE_PROPERTIES`